src/services/navigation/useLocationParams.ts

```diff
@@ -2,7 +2,7 @@ import omitBy from 'lodash/omitBy'
 import qs from 'qs'
 import { useHistory, useLocation } from 'react-router-dom'
 
-export function useLocationParams(defaultParams = {}) {
+export function useLocationParams(defaultParams: Record<string, unknown> = {}) {
```

Comment:
I wonder if we can use some TS generics here so that the default params that we pass in are typed to the return value of `params`

Reply:
And i wonder if we can utilize these same generics to the update functions etc. so that when we call them we know the keys, and their value types.

Reply:
I think I had it most of the way there in this [file](https://github.com/codecov/gazebo/pull/3439/commits/5fb27b51aab067ff14276161a9a7cccffd694055#diff-4de4e08bfe523e70ad444de12031f936e33b5766704c977d67b9944012b718d8) but the changes it touched started to explode to 10+ other files and fixes needed. Spun it into this ticket and I can take another stab later - https://github.com/codecov/engineering-team/issues/2843

Reply:
Okay sounds good

Reply:
If it's gonna be a significant investment, we may want to skip it, and wait until we move routers to save time ... and potentially lots of hair pulling.